### PR TITLE
Better batch endpoint: accepts multiply texts, returns also segmented…

### PR DESCRIPTION
**Pull request by dchaplinsky**
… and cleansed version of the text alongside with tokenized and lemmatized

Please also take a look at the questions I left in the comments

### CODE REVIEW

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy [hunk 1]
```diff
@@ -39,14 +50,21 @@ class BatchController extends BaseController {
...
         validateRequest(request)
 
         try {
-            batchService.batch(request.text)
+            // To Andriy. Can I use parallel here? groovy gave me the error when compiling.
+            // Maybe it is related to the old version of it?
+            // Groovy Version: 4.0.5 JVM: 17.0.4.1 Vendor: Homebrew OS: Mac OS X
+            // It's not a mandatory option, so if it's hard to implement, I'd rather skip this one
+            // return request.texts.parallel().collect { text ->
+            return request.texts.collect { text ->
```
Comment (arysin) [inline app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: You can try Java parallel stream + (simplified) Groovy collect:
```
request.tests.parallelStream().collect{ 
  batchService.batch(text)
}
```

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy [hunk 2]
```diff
@@ -57,13 +75,18 @@ class BatchController extends BaseController {
...
     }
 
+
     @Schema(description="Response with tokenized and lemmatized text.")
     @CompileStatic
     static class BatchResponse {
+        String clean
         List<List<String>> tokenized
         List<List<String>> lemmatized
+        List<String> segmented
```
Comment (arysin) [inline app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: I wonder if should call this 'sentences' (and the other two 'tokens' and 'lemmas'?

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy [hunk 3]
```diff
@@ -57,13 +75,18 @@ class BatchController extends BaseController {
...
+    static class BatchRequest {
+        @Schema(description="Texts to process.")
+        List<String> texts
     }
 
+
     @Schema(description="Response with tokenized and lemmatized text.")
     @CompileStatic
     static class BatchResponse {
+        String clean
```
Comment (arysin) [inline app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: I'd call it "cleanText"

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy (hunk 1 above)
Comment (dchaplinsky) [reply app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: lang-uk/nlp_uk_api/app/src/main/groovy/ua/net/nlp/api/BatchController.groovy: 65: [Static type checking] - Cannot find matching method java.util.List#parallelStream(groovy.lang.Closure). Please check if the declared type is correct and if the method exists.
 @ line 65, column 20.
               return request.texts.parallelStream() { text ->

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy [hunk 4]
```diff
@@ -29,8 +29,19 @@ class BatchController extends BaseController {
     @Autowired
     BatchService batchService
 
+    def validateRequest(BatchRequest request) {
```
Comment (dchaplinsky) [inline app/src/main/groovy/ua/net/nlp/api/BatchController.groovy:36]: Could you also please comment on this one?

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy (hunk 1 above)
Comment (arysin) [reply app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: `texts.parallelStream().collect{ ... }`

Comment (dchaplinsky) [reply app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: /lang-uk/nlp_uk_api/app/src/main/groovy/ua/net/nlp/api/BatchController.groovy: 66: [Static type checking] - Cannot find matching method ua.net.nlp.api.services.BatchService#batch(java.lang.Object). Please check if the declared type is correct and if the method exists.
 @ line 66, column 17.
                   batchService.batch(text)
                   ^

1 error


```groovy
            return request.texts.parallelStream().collect { text ->
                batchService.batch(text)
            }

```

Comment (arysin) [reply app/src/main/groovy/ua/net/nlp/api/BatchController.groovy]: `parallelStream().collect{ String text ->`

File: app/src/main/groovy/ua/net/nlp/api/BatchController.groovy (hunk 4 above)
Comment (arysin) [reply app/src/main/groovy/ua/net/nlp/api/BatchController.groovy:36]:
```
request.texts.each { 
if( ! it ) ... 
if( it.length() > .... ) ...
}
```

**Pull request merged**